test/distributed/_composable/fsdp/test_fully_shard_training.py

```diff
@@ -273,19 +294,25 @@ def _test_train_parity_multi_group(
         orig_reduce_scatter = dist.reduce_scatter_tensor
 
         def delayed_all_gather(*args, **kwargs):
-            if delay_before_all_gather:
-                torch.cuda._sleep(int(delay_in_ms * get_cycles_per_ms()))
+            torch.cuda._sleep(int(delay_in_ms * get_cycles_per_ms()))
```

Comment:
Curious why this change -- only stylistic or it affects compile?

Reply:
same to above. ``delayed_all_gather`` does not work in compiling. need to figure out the root cause